solutions/cf-977F.mdx

```diff
@@ -66,4 +64,69 @@ public class ConsecutiveSubsequence {
 
 </JavaSection>
 
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+#include <map>
+#include <algorithm>
+ 
+using std::cout;
+using std::endl;
+using std::vector;
+using std::pair;
+ 
+/**
+ * https://codeforces.com/contest/977/problem/F
+ * 7
+ * 3 3 4 7 5 6 8
+ */
+int main() {
+	int size;
+	std::cin >> size;
+	vector<int> arr(size);
+	for (int& i : arr) {
+		std::cin >> i;
+	}
+ 
+	int best_len = 1;
+	// given the ending limit, stores the longest consecutive sequence
+	std::map<int, pair<int, int>> ending_max_len;
+	std::map<int, int> come_from;
```

Comment:
(not necessary) add comments if you can? i'd comment but i forgot how this problem worked ahahaha

Reply:
bump